### PR TITLE
Add password dialog option to Notifier

### PR DESCRIPTION
**Pull request by kuan9611**
Issue: [#1431](https://github.com/mit-cml/appinventor-sources/issues/1431)
Method for Notifier to prompt for password rather than plain text.

### CODE REVIEW

**Comment by josmas** [comment]
@kuan9611 thanks for the PR
@ewpatton does this need a [change in YaVersion](https://docs.google.com/document/d/1ukpZXcd0fIn1fV0cNp4J00wrfwwA1daswU1yY6dqqfs/pub#h.2g0jyctaq6go)?

**Comment by ewpatton** [comment]
@josmas Yes that's true.
@kuan9611 You'll want to make the following changes:
- [ ] Bump the `NOTIFIER_COMPONENT_VERSION` in YaVersion.java (and add associated comment)
- [ ] Bump the `YOUNG_ANDROID_VERSION` in YaVersion.java (and add associated comment)
- [ ] In YoungAndroidFormUpgrader.java, add a branch to upgradeNotifierProperties. Since no upgrade is needed, just increase the version number
- [ ] In versioning.js, add an entry to the Notifier branch of Blockly.Versioning.AllUpgradeMaps for the version number to be "noUpgrade" (as we are adding a new block).

Also, in the future you should make a new branch and push there rather than your master branch.

**Comment by josmas** [comment]
@kuan9611 once you have the new changes, simply commit and push again. It will show up here as an added commit.

**Comment by josmas** [comment]
As it is, the PR builds and works fine on Android P.

**Comment by kuan9611** [comment]
@ewpatton Updated the version files. I'll make sure to push to a new branch in the future.

File: appinventor/appengine/src/com/google/appinventor/client/youngandroid/YoungAndroidFormUpgrader.java [hunk 1]
```diff
@@ -1368,6 +1368,10 @@ private static int upgradeNotifierProperties(Map<String, JSONValue> componentPro
       // Added TextInputCanceled & ChoosingCanceled event
       srcCompVersion = 5;
     }
+    if (srcCompVersion < 6) {
```
Comment (josmas) [inline appinventor/appengine/src/com/google/appinventor/client/youngandroid/YoungAndroidFormUpgrader.java]: note indentation here is wrong. The project uses 2 spaces for both java and javascript.

Comment (kuan9611) [reply appinventor/appengine/src/com/google/appinventor/client/youngandroid/YoungAndroidFormUpgrader.java]: oops I'll fix that quickly

**Comment by josmas** [comment]
After changes, it still works nicely on P. Only issue is indentation (comment in code).

My emulator is not working on mac, I have to install the new package. @kuan9611 have you tried it on it?

**Comment by kuan9611** [comment]
@josmas No but I tried it on my Android phone with no issues.

**Comment by josmas** [comment]
@kuan9611 which version of Android are you running? we need as much coverage as possible, especially on older devices.

**Comment by kuan9611** [comment]
@josmas 6.0 marshmellow

**Comment by ewpatton** [comment]
I tested with an emulator running SDK 10 and it worked okay. I will test on the Windows emulator later tonight to confirm it works that far back (but I expect it should be okay).

**Comment by kuan9611** [comment]
@ewpatton sounds good. I've added to the documentation.

**Comment by ewpatton** [comment]
Tested on Android 7 emulator and it works there.

**Pull request merged**